src/core.js

```diff
@@ -1300,12 +1300,12 @@ $.extend( $.validator, {
 			this.settings.messages[ element.name ].remote = previous.message;
 
 			param = typeof param === "string" && { url: param } || param;
-
-			if ( previous.old === value ) {
+			optionDataString = JSON.stringify( $.param( $.extend( { data: value }, param.data ) ) );
```

Comment:
Unless you think it's more correct to serialize the json object, I just pictured seeing a data property containing a JSON object and thought that would be bad or confusing vs seeing the querystring that was last sent to the endpoint, which makes more sense to me.